client/src/components/Slider/PageSlider.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactNode, useEffect, useRef, useState } from 'react';
+import { SLIDER_POSITION } from 'src/constants/sliderPosition';
+import styled from 'styled-components/macro';
+
+interface PageSliderProps {
+  children: ReactNode;
+  animationCallback: any;
+  pagePosition: string;
+}
+
+export default function PageSlider({
+  children,
+  animationCallback,
+  pagePosition,
+}: PageSliderProps) {
+  const [position, setPosition] = useState('');
+  const sliderRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const startAnimation = () => {
+      setPosition(pagePosition);
+    };
+
+    const onTransitionEnd = (e: any) => {
+      if (e.propertyName !== 'transform') return;
+      animationCallback();
+    };
+
+    if (sliderRef.current) {
+      sliderRef.current.addEventListener('transitionend', onTransitionEnd);
+      startAnimation();
+    }
+  }, [children]);
+
+  const getClassName = () => {
+    let className = `animatable`;
+    if (SLIDER_POSITION.TO_LEFT === pagePosition)
+      className += ' prepare-to-left';
+    if (SLIDER_POSITION.FROM_RIGHT === pagePosition)
+      className += ' prepare-from-rigth';
+    if (SLIDER_POSITION.TO_LEFT === position) className += ' to left';
+    if (SLIDER_POSITION.FROM_RIGHT === position) className += ' from right';
+    return className;
+  };
+
+  return (
+    <Container>
+      <div ref={sliderRef} className={getClassName()}>
+        {children}
+      </div>
+    </Container>
+  );
+}
+
+const Container = styled.div`
+  .animatable {
+    background: inherit;
+    will-change: transform;
+    transition: transform 10s;
+  }
```

Comment:
Container 스타일 컴포넌트는 이런 형태가 될 것 같아요. 

[공식 문서](https://styled-components.com/docs/basics#pseudoelements-pseudoselectors-and-nesting)에 나와있는 props 예제를 참고했는데, 이 방식은 어떻게 생각하시나요?

```suggestion
const Container = styled.div<{ pagePosition: string }>`
  ${({ pagePosition }) => {
    switch (pagePosition) {
      case SLIDER_POSITION.TO_LEFT:
        return css`
          transform: translateX(-1px);
        `;
    }
  }}
```

Reply:
이렇게 좋은게 있었군요!!!!!!!!

Reply:
예시까지 들어서 설명해주셔서 감사해요 수정해서 푸쉬했습니다